src/main/java/com/fdmgroup/forex/services/FundTransferService.java

```diff
@@ -0,0 +1,60 @@
+package com.fdmgroup.forex.services;
+
+import java.util.Date;
+import java.util.UUID;
+
+import org.springframework.stereotype.Service;
+
+import com.fdmgroup.forex.enums.TransferType;
+import com.fdmgroup.forex.models.FundTransfer;
+import com.fdmgroup.forex.repos.FundTransferRepo;
+
+@Service
+public class FundTransferService {
+	private FundTransferRepo fundTransferRepo;
+	private CurrencyService currencyService;
+	private PortfolioService portfolioService;
+	private AssetService assetService;
+
+	public FundTransferService(FundTransferRepo fundTransferRepo, CurrencyService currencyService, PortfolioService portfolioService, AssetService assetService) {
+		this.fundTransferRepo = fundTransferRepo;
+		this.currencyService = currencyService;
+		this.portfolioService = portfolioService;
+		this.assetService = assetService;
+	}
+	
+	public FundTransfer transferFund(FundTransfer fundTransfer) {
+		getManagedAttributes(fundTransfer);
+		fundTransfer.setTransferDate(new Date());
+		if (fundTransfer.getTransferType().equals(TransferType.DEPOSIT)) {
+			deposit(fundTransfer);
+		} else {
+			withdrawal(fundTransfer);
+		}
+		return fundTransferRepo.save(fundTransfer);
+	}
+	
+	private void deposit(FundTransfer fundTransfer) {
+		assetService.depositAsset(fundTransfer.getPortfolio(),fundTransfer.getCurrency(),fundTransfer.getAmount());
+	}
+	
+	private void withdrawal(FundTransfer fundTransfer) throws RecordNotFoundException, InsufficientFundsException {
+		assetService.withdrawAsset(fundTransfer.getPortfolio(),fundTransfer.getCurrency(),fundTransfer.getAmount());
+	}
+
+
+	public void getManagedAttributes(FundTransfer fundTransfer) {
+		getManagedCurrency(fundTransfer);
+		getManagedPortfolio(fundTransfer);
+	}
+
+	private void getManagedCurrency(FundTransfer fundTransfer) {
```

Comment:
Would a better method name be `validateCurrency()`? It appears to me that you're using the call to `currencyService` to confirm that the currency exists, then re-setting the same currency for the `fundTransfer` - if it's not found it will through an exception.

If that's the case then "validate" would more accurately reflect this behaviour.